src/app/workflow-editor/graph-editor/graph-editor/workflow-graph-editor.component.ts

```diff
@@ -353,10 +344,43 @@ export class WorkflowGraphEditorComponent extends DirectiveBase implements OnCha
             const input        = this.model.inputs.find((input) => connectionId === input.connectionId);
             const output       = this.model.outputs.find((output) => connectionId === output.connectionId);
             this.inspectedNode = step || input || output;
+
+            // When you create some node (i/o or step by dropping it on a canvas) and open it in object inspector, when
+            // you go backward in history (undo) object inspector should be closed
+            if (!this.inspectedNode) this.inspector.hide();
         }
     }
 
-    ngOnChanges() {
+    /**
+     * Register event listeners on a current model
+     */
+    registerModelEventListeners() {
+        // Close object inspector if step/input/output is removed
+        const removeHandler = (node) => {
+            if (this.inspectedNode && this.inspectedNode.id === node.id) {
+                this.inspector.hide();
+                this.inspectedNode = null;
+            }
+        };
+
+        this.modelEventListeners = [
+            this.model.on("output.remove", removeHandler),
```

Comment:
You can set all of these to the `DirectiveBase.tracked` property and they will automatically be cleaned up on `ngOnDestroy`. That's actually why I named the method `dispose`, so that `tracked` would pick it up and remove it.